buffer/src/main/java/io/netty/buffer/UnpooledDirectByteBuf.java

```diff
@@ -39,6 +40,7 @@ public class UnpooledDirectByteBuf extends AbstractReferenceCountedByteBuf {
 
     private final ByteBufAllocator alloc;
 
+    CleanableDirectBuffer cleanable;
     ByteBuffer buffer; // accessed by UnpooledUnsafeNoCleanerDirectByteBuf.reallocateDirect()
```

Comment:
Do we still need this ? can't we access it via the cleanable ?

Reply:
We can, but it adds an indirection (and makes the change bigger)

Reply:
I am just a bit concerned that we add just another field and add more overhead (memory).

Reply:
In the common case yeah, this does increase the object size by 8 bytes… but we don't always have a cleanable, so then we'd have to wrap it in a noop implementation